mypy/suggestions.py

```diff
@@ -404,6 +404,22 @@ def find_best(self, func: FuncDef, guesses: List[CallableType]) -> Tuple[Callabl
                    key=lambda s: (count_errors(errors[s]), self.score_callable(s)))
         return best, count_errors(errors[best])
 
+    def get_guesses_from_parent(self, node: FuncDef) -> List[CallableType]:
+        """Try to get a guess of a method type from a parent class."""
+        if not node.info:
+            return []
+
+        for parent in node.info.mro[1:]:
+            pnode = parent.names.get(node.name)
+            if pnode and isinstance(pnode.node, (FuncDef, Decorator)):
+                typ = pnode.node.type
+                if isinstance(typ, CallableType) and len(typ.arg_types) == len(node.arguments):
+                    # Return the first thing we find, since it probably doesn't make sense
+                    # to grab things further up in the chain if an earlier parent has it.
+                    return [pnode.node.type]
```

Comment:
This doesn't work as expected if a base class is generic. Maybe fix it (if it's easy to fix) or add a TODO comment.